redbot/cogs/mutes/mutes.py

```diff
@@ -0,0 +1,1212 @@
+                del self._server_mutes[g_id][u_id]
+            if self._server_mutes[g_id] == {}:
+                to_clear.append(g_id)
+            await self.config.guild(guild).muted_users.set(self._server_mutes[g_id])
+        for g_id in to_clear:
+            await self.config.guild_from_id(g_id).muted_users.clear()
+
+    async def _auto_unmute_user(self, guild: discord.Guild, data: dict):
+        delay = data["until"] - datetime.now(timezone.utc).timestamp()
+        if delay < 1:
+            delay = 0
+        await asyncio.sleep(delay)
+        try:
+            member = guild.get_member(data["member"])
+            author = guild.get_member(data["author"])
+            if not member:
+                return
+            success, message = await self.unmute_user(guild, author, member, _("Automatic unmute"))
+            if success:
+                try:
+                    await modlog.create_case(
+                        self.bot,
+                        guild,
+                        datetime.now(timezone.utc),
+                        "sunmute",
+                        member,
+                        author,
+                        _("Automatic unmute"),
+                        until=None,
+                    )
+                except RuntimeError as e:
+                    log.error(_("Error creating modlog case"), exc_info=e)
+        except discord.errors.Forbidden:
+            return
+
+    async def _handle_channel_unmutes(self):
+        log.debug("Checking channel unmutes")
+        to_clear = []
+        for c_id, mutes in self._channel_mutes.items():
+            to_remove = []
+            channel = self.bot.get_channel(c_id)
+            if channel is None:
+                continue
+            for u_id, data in mutes.items():
+                if not data or not data["until"]:
+                    continue
+                time_to_unmute = data["until"] - datetime.now(timezone.utc).timestamp()
+                if time_to_unmute < 120.0:
+                    self._unmute_tasks[f"{c_id}{u_id}"] = asyncio.create_task(
+                        self._auto_channel_unmute_user(channel, data)
+                    )
+            for u_id in to_remove:
+                del self._channel_mutes[c_id][u_id]
+            if self._channel_mutes[c_id] == {}:
+                to_clear.append(c_id)
+            await self.config.channel(channel).muted_users.set(self._channel_mutes[c_id])
+        for c_id in to_clear:
+            await self.config.channel_from_id(c_id).muted_users.clear()
+
+    async def _auto_channel_unmute_user(self, channel: discord.TextChannel, data: dict):
+        delay = data["until"] - datetime.now(timezone.utc).timestamp()
+        if delay < 1:
+            delay = 0
+        await asyncio.sleep(delay)
+        try:
+            member = channel.guild.get_member(data["member"])
+            author = channel.guild.get_member(data["author"])
+            if not member or not author:
```

Comment:
You probably meant to only include the mute's target, not mute's author in this if:
```suggestion
            if not member:
```